From 555a57fc16c0460dc08ba41de469b503d437dbbf Mon Sep 17 00:00:00 2001
From: Octavian Purdila <octavian.purdila@nxp.com>
Date: Wed, 22 Feb 2017 17:32:46 +0200
Subject: [PATCH 1435/5242] MLK-13869 ARM: dts: imx7: route interrupts through
 GPC

commit  fd8f265c24f7e2ee3b28b55a0b3110d61f8fb33a from
https://source.codeaurora.org/external/imx/linux-imx.git

We need to "logically" route interrupts through GPC instead of directly
through GIC in order to support low power mode with SCU and L2 off.

Signed-off-by: Octavian Purdila <octavian.purdila@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/imx7d.dtsi |   30 ------------------------------
 arch/arm/boot/dts/imx7s.dtsi |    5 +++++
 2 files changed, 5 insertions(+), 30 deletions(-)

diff --git a/arch/arm/boot/dts/imx7d.dtsi b/arch/arm/boot/dts/imx7d.dtsi
index fe96d38..14fe736 100644
--- a/arch/arm/boot/dts/imx7d.dtsi
+++ b/arch/arm/boot/dts/imx7d.dtsi
@@ -185,42 +185,12 @@
 		compatible = "fsl,imx6q-caam-snvs";
 		reg = <0x30370000 0x10000>;
 	};
-
-	gpc: gpc@303a0000 {
-		compatible = "fsl,imx7d-gpc";
-		reg = <0x303a0000 0x10000>;
-		interrupt-controller;
-		interrupts = <GIC_SPI 87 IRQ_TYPE_LEVEL_HIGH>;
-		#interrupt-cells = <3>;
-		interrupt-parent = <&intc>;
-		fsl,mf-mix-wakeup-irq = <0x54010000 0xc00 0x0 0x1040640>;
-		mipi-phy-supply = <&reg_1p0d>;
-		pcie-phy-supply = <&reg_1p0d>;
-		vcc-supply = <&reg_1p2>;
-	};
-
 	iomuxc_lpsr_gpr: lpsr-gpr@30270000 {
 		compatible = "fsl,imx7d-lpsr-gpr";
 		reg = <0x30270000 0x10000>;
 	};
 };
 
-&anatop {
-	reg_1p2: regulator-vdd1p2@220 {
-		compatible = "fsl,anatop-regulator";
-		regulator-name = "vdd1p2";
-		regulator-min-microvolt = <1100000>;
-		regulator-max-microvolt = <1300000>;
-		anatop-reg-offset = <0x220>;
-		anatop-vol-bit-shift = <8>;
-		anatop-vol-bit-width = <5>;
-		anatop-min-bit-val = <8>;
-		anatop-min-voltage = <1100000>;
-		anatop-max-voltage = <1300000>;
-		anatop-enable-bit = <31>;
-	};
-};
-
 &aips2 {
 	flextimer1: flextimer@30640000 {
 		compatible = "fsl,imx7d-flextimer";
diff --git a/arch/arm/boot/dts/imx7s.dtsi b/arch/arm/boot/dts/imx7s.dtsi
index 339c65a..fb9da11 100644
--- a/arch/arm/boot/dts/imx7s.dtsi
+++ b/arch/arm/boot/dts/imx7s.dtsi
@@ -154,6 +154,7 @@
 			     <GIC_PPI 14 (GIC_CPU_MASK_SIMPLE(4) | IRQ_TYPE_LEVEL_LOW)>,
 			     <GIC_PPI 11 (GIC_CPU_MASK_SIMPLE(4) | IRQ_TYPE_LEVEL_LOW)>,
 			     <GIC_PPI 10 (GIC_CPU_MASK_SIMPLE(4) | IRQ_TYPE_LEVEL_LOW)>;
+		clock-frequency = <8000000>;
 	};
 
 	soc {
@@ -607,6 +608,10 @@
 				#interrupt-cells = <3>;
 				interrupt-parent = <&intc>;
 				#power-domain-cells = <1>;
+				fsl,mf-mix-wakeup-irq = <0x54010000 0xc00 0x0 0x1040640>;
+				mipi-phy-supply = <&reg_1p0d>;
+				pcie-phy-supply = <&reg_1p0d>;
+				vcc-supply = <&reg_1p2>;
 
 				pgc {
 					#address-cells = <1>;
-- 
1.7.9.5

